File: Frontend/src/app/models/payload/maker-search-payload.ts

export class MakerSearchPayload {

   branch_id?: number;
   tax_category_id?: number;
   reference_number?: string;
   status_id?: Number;
   user_id?: number;
   drafted_date?: Date[];
   maker_date?: Date[];
   checked_date?: Date[];
   rejected_date?: Date[];
   approved_date?: Date[];
   branch_name?: String
   routeControl?: string;



//  paginator payload   
    totalRecords: number = 0;
  currentPage?: number = 0; 
  pageSize?: number = 10;
  event_length?: number = 0;

}

File: Backend/TaxManagementSystem/src/main/java/com/tms/Maker/entity/MakerSearchPayload.java
package com.tms.Maker.entity;

import java.sql.Date;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data

@AllArgsConstructor

@NoArgsConstructor

public class MakerSearchPayload {

    private Long id;
    private String routeControl;
    private String user_name;
    private int status;

    private Long branch_id;
    private Long tax_category_id;
    private String reference_number;
    private Long status_id;
    private Date maked_date;
    private String document_type;
    private Long user_id;
     @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private List<Date> drafted_date; 
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
    private List<Date> maker_date; 
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")
     private List<Date> checked_date;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")

     private List<Date> rejected_date;
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd")

     private List<Date> approved_date;



}
